bin/compile

```diff
@@ -51,7 +51,17 @@ function vendor() {
 
 echo "Vendoring binaries" | arrow
 
-vendor "${ROOT_DIR}/build/imagemagick-heroku-20.tar.gz" "$BUILD_DIR/vendor/imagemagick"
+if [[ "$STACK" == "heroku-24" ]]; then
+  echo "Unsupported stack: $STACK"
+  exit 1
+elif [[ "$STACK" == "heroku-22" ]]; then
+  vendor "${ROOT_DIR}/build/imagemagick-heroku-22.tar.gz" "$BUILD_DIR/vendor/imagemagick"
+elif [[ "$STACK" == "heroku-20" ]]; then
+  vendor "${ROOT_DIR}/build/imagemagick-heroku-20.tar.gz" "$BUILD_DIR/vendor/imagemagick"
```

Comment:
Merging this PR should not affect production or any of our Heroku apps that are running Heroku-20.